CORE/world_gadgets/router/page_spec.rb

```diff
@@ -33,7 +24,7 @@ def inherit_details(parent_page)
   end
 
   def add_route(target_page, action, prerequisites)
-    @routes[target_page] ||= Route.new( target_page, action, prerequisites )
+    @routes[target_page] ||= Route.new( target_page, action, [ prerequisites ].flatten )
```

Comment:
It was being flattened before in the Route class (line 9 above), but by changing that class to be a Struct we need to make sure that happens before the data gets pushed into that Struct instead. 

The reason this needs to happen at all is because this argument passed to `add_route` can be either a symbol or an array of symbols (because you may have 0, 1, or many prerequisites) and this will always make sure that it is an array (of one dimension) before being saved in the Struct